lib/src/gestures/flutter_map_interactive_viewer.dart

```diff
@@ -0,0 +1,856 @@
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_map/src/gestures/interactive_flag.dart';
+import 'package:flutter_map/src/gestures/latlng_tween.dart';
+import 'package:flutter_map/src/gestures/map_events.dart';
+import 'package:flutter_map/src/gestures/multi_finger_gesture.dart';
+import 'package:flutter_map/src/map/flutter_map_frame.dart';
+import 'package:flutter_map/src/map/flutter_map_internal_controller.dart';
+import 'package:flutter_map/src/map/flutter_map_internal_state.dart';
+import 'package:flutter_map/src/map/options.dart';
+import 'package:flutter_map/src/misc/point.dart';
+import 'package:flutter_map/src/misc/private/positioned_tap_detector_2.dart';
+import 'package:latlong2/latlong.dart';
+
+class FlutterMapInteractiveViewer extends StatefulWidget {
+  final Widget Function(BuildContext context, FlutterMapInternalState mapState)
+      builder;
+  final FlutterMapInternalController controller;
+
+  const FlutterMapInteractiveViewer({
+    super.key,
+    required this.builder,
+    required this.controller,
+  });
+
+  @override
+  State<FlutterMapInteractiveViewer> createState() =>
+      FlutterMapInteractiveViewerState();
+}
+
+class FlutterMapInteractiveViewerState
+    extends State<FlutterMapInteractiveViewer> with TickerProviderStateMixin {
+  static const int _kMinFlingVelocity = 800;
+  static const _kDoubleTapZoomDuration = 200;
+
+  final _positionedTapController = PositionedTapController();
+  final _gestureArenaTeam = GestureArenaTeam();
+  late Map<Type, GestureRecognizerFactory> _gestures;
+
+  bool _dragMode = false;
+  int _gestureWinner = MultiFingerGesture.none;
+  int _pointerCounter = 0;
+  bool _isListeningForInterruptions = false;
+
+  var _rotationStarted = false;
+  var _pinchZoomStarted = false;
+  var _pinchMoveStarted = false;
+  var _dragStarted = false;
+  var _flingAnimationStarted = false;
+
+  // Helps to reset ScaleUpdateDetails.scale back to 1.0 when a multi finger
+  // gesture wins
+  late double _scaleCorrector;
+  late double _lastRotation;
+  late double _lastScale;
+  late Offset _lastFocalLocal;
+  late LatLng _mapCenterStart;
+  late double _mapZoomStart;
+  late Offset _focalStartLocal;
+  late LatLng _focalStartLatLng;
+
+  late final AnimationController _flingController;
```

Comment:
Prefer assigning the controller directly on the declaration line to make it more readable (listeners can be kept inside the `initState`):
```dart
late final AnimationController _flingController =  AnimationController(vsync: this);
```